tensorflow_lite_support/cc/task/vision/utils/image_tensor_specs.cc

```diff
@@ -132,55 +141,67 @@ StatusOr<absl::optional<NormalizationOptions>> GetNormalizationOptionsIfAny(
   return normalization_options;
 }
 
-}  // namespace
-
-StatusOr<ImageTensorSpecs> BuildInputImageTensorSpecs(
+StatusOr<ImageTensorSpecs> BuildImageTensorSpecs(
     const TfLiteEngine::Interpreter& interpreter,
-    const tflite::metadata::ModelMetadataExtractor& metadata_extractor) {
+    const tflite::metadata::ModelMetadataExtractor& metadata_extractor,
+    const bool is_input) {
   ASSIGN_OR_RETURN(const TensorMetadata* metadata,
-                   GetInputTensorMetadataIfAny(metadata_extractor));
+                   GetTensorMetadataIfAny(metadata_extractor, is_input));
 
   const ImageProperties* props = nullptr;
   absl::optional<NormalizationOptions> normalization_options;
   if (metadata != nullptr) {
     ASSIGN_OR_RETURN(props, GetImagePropertiesIfAny(*metadata));
     ASSIGN_OR_RETURN(normalization_options,
                      GetNormalizationOptionsIfAny(*metadata));
+    if (!is_input && !normalization_options.has_value()) {
+      ASSIGN_OR_RETURN(normalization_options,
+                       GetNormalizationOptionsIfAny(
+                           *metadata_extractor.GetInputTensorMetadata(0)));
+    }
   }
 
-  if (TfLiteEngine::InputCount(&interpreter) != 1) {
+  if (is_input && TfLiteEngine::InputCount(&interpreter) != 1) {
     return CreateStatusWithPayload(
         StatusCode::kInvalidArgument,
         "Models are assumed to have a single input.",
         TfLiteSupportStatus::kInvalidNumInputTensorsError);
+  } else if (!is_input && TfLiteEngine::OutputCount(&interpreter) != 1) {
+    return CreateStatusWithPayload(
+        StatusCode::kInvalidArgument,
+        "Models are assumed to have a single output.",
+        TfLiteSupportStatus::kInvalidNumOutputTensorsError);
   }
 
-  // Input-related specifications.
-  const TfLiteTensor* input_tensor = TfLiteEngine::GetInput(&interpreter, 0);
-  if (input_tensor->dims->size != 4) {
+  // Input/output-related specifications.
+  const TfLiteTensor* tensor = is_input
+                                   ? TfLiteEngine::GetInput(&interpreter, 0)
```

Comment:
Again, you need to pass TfLiteTensor into BuildImageTensorSpecs, because it's not always the first Input or output tensor.